test/ArcxAnalyticsProvider.test.tsx

```diff
@@ -0,0 +1,150 @@
+import { cleanup, render, screen } from '@testing-library/react'
+import {
+  ArcxAnalyticsProvider,
+  ArcxAnalyticsProviderProps,
+  ArcxAnalyticxContext,
+  useArcxAnalytics,
+} from '../src'
+import { expect } from 'chai'
+import sinon from 'sinon'
+import * as postRequestModule from '../src/helpers/postRequest'
+import { ATTRIBUTION_EVENT, DEFAULT_SDK_CONFIG } from '../src/constants'
+import React from 'react'
+
+const IDENTITY_ID = 'test-dentity-id'
+const TEST_API_KEY = 'test-api-key'
+const CONFIG = {}
+
+const TestProvider = ({
+  children,
+  providerOverrides,
+}: {
+  children?: React.ReactNode
+  providerOverrides?: ArcxAnalyticsProviderProps
+}) => (
+  <ArcxAnalyticsProvider apiKey={TEST_API_KEY} config={CONFIG} {...providerOverrides}>
+    <ArcxAnalyticxContext.Consumer>
+      {(sdk) => (
+        <div>
+          <div>{sdk?.identityId}</div>
+          <div>{children}</div>
+        </div>
+      )}
+    </ArcxAnalyticxContext.Consumer>
+  </ArcxAnalyticsProvider>
+)
+
+const ChildTest = () => {
+  const sdk = useArcxAnalytics()
+
+  return (
+    <div>
+      <button onClick={() => sdk?.page({ url: '/test' })}>fire page event</button>
+      <button onClick={() => sdk?.event('test-event', { gm: 'gm' })}>fire custom event</button>
+      <button onClick={() => sdk?.transaction({ chain: 1, transactionHash: '0x123' })}>
+        fire transaction event
+      </button>
+      <button
+        onClick={() =>
+          sdk?.attribute({ source: 'facebook', medium: 'social', campaign: 'ad-camp' })
+        }
+      >
+        fire attribute event
+      </button>
+    </div>
+  )
+}
+
+describe('ArcxAnalyticxProvider', () => {
+  let postRequestStub: sinon.SinonStub
+
+  beforeEach(async () => {
+    postRequestStub = sinon.stub(postRequestModule, 'postRequest').resolves(IDENTITY_ID)
+    render(<ChildTest />, { wrapper: TestProvider })
+    expect(await screen.findByText(IDENTITY_ID)).to.exist
+    postRequestStub.resetHistory()
+  })
+
+  afterEach(sinon.restore)
+
+  it('initializes the SDK', async () => {
+    cleanup()
```

Comment:
why do we call cleanup before each test and not `afterEach(() => `

Reply:
To clear the `screen` object, otherwise the HTML rendered with `render()` persists between tests

Reply:
That's why we need to clean it after we render, not before?

Reply:
Not really because some rendering is done in the `beforeEach`, where others need to render something from scratch

Reply:
I got you point- we can leave it here. But still, between tests we need to have clean state of stubs/renders/mocks to avoid some strange behaviour, so it should be added as well in afterEach, or somewhere at the end

Reply:
I like it, adjusted.